src/components/forms/pool_actions/MigrateForm/composables/useMigrateMath.ts

```diff
@@ -0,0 +1,133 @@
+import { computed, ref, Ref } from 'vue';
+import { parseUnits } from 'ethers/lib/utils';
+
+import useNumbers from '@/composables/useNumbers';
+import useTokens from '@/composables/useTokens';
+import useUserSettings from '@/composables/useUserSettings';
+
+import { FullPool } from '@/services/balancer/subgraph/types';
+import PoolCalculator from '@/services/pool/calculator/calculator.sevice';
+
+import { bnSum, bnum } from '@/lib/utils';
+
+export type MigrateMathResponse = ReturnType<typeof useMigrateMath>;
+
+export default function useMigrateMath(
+  fromPool: Ref<FullPool>,
+  toPool: Ref<FullPool>
+) {
+  /**
+   * COMPOSABLES
+   */
+  const { tokens, balances, balanceFor, getToken } = useTokens();
+  const { currency } = useUserSettings();
+  const { fNum, toFiat } = useNumbers();
+
+  /**
+   * SERVICES
+   */
+  const fromPoolCalculator = new PoolCalculator(
+    fromPool,
+    tokens,
+    balances,
+    'exit',
+    ref(false)
+  );
+
+  const toPoolCalculator = new PoolCalculator(
+    toPool,
+    tokens,
+    balances,
+    'join',
+    ref(false)
+  );
+
+  /**
+   * COMPUTED
+   */
+  const bptBalance = computed(() => balanceFor(fromPool.value.address));
+
+  const hasBpt = computed(() => bnum(bptBalance.value).gt(0));
+
+  const tokenCount = computed(() => fromPool.value.tokenAddresses.length);
+
+  const poolDecimals = computed(() => fromPool.value.onchain.decimals);
+
+  const bptBalanceScaled = computed(() =>
+    parseUnits(bptBalance.value, poolDecimals.value).toString()
+  );
+
+  const withdrawalTokens = computed(() =>
+    fromPool.value.tokenAddresses.map(address => getToken(address))
+  );
+
+  const fullAmounts = computed(() => {
+    const { receive } = fromPoolCalculator.propAmountsGiven(
+      bptBalance.value,
+      0,
+      'send'
+    );
+    return receive;
+  });
+
+  const fullAmountsScaled = computed((): string[] =>
+    fullAmounts.value.map((amount, i) =>
+      parseUnits(amount, withdrawalTokens.value[i].decimals).toString()
+    )
+  );
+
+  const fullBPTOut = computed(() =>
+    fromPoolCalculator.exactTokensInForBPTOut(fullAmounts.value).toString()
+  );
```

Comment:
This is what I think is incorrect, this should be the bpt out from the bb-a-USD pool. To get that you will need to use a SDK static call to the batch relayer as we do in useWithdrawalMath. You may also need to make the UI wait for this to fetch too.